timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimeLockConfigMigrator.java

```diff
@@ -24,34 +23,54 @@ private TimeLockConfigMigrator() { /* Utility Class */ }
-                        .dataDirectory(paxos.paxosDataDir())
-                        .build())
-                .cluster(ImmutableClusterConfiguration.builder()
-                        .cluster(ServiceConfiguration.builder()
-                                .security(paxos.sslConfiguration())
-                                .uris(config.cluster().servers())
-                                .build())
-                        .localServer(config.cluster().localServer())
-                        .build())
-                .asyncLock(config.asyncLockConfiguration())
-                .build();
+            runtime = ImmutableTimeLockRuntimeConfiguration.builder()
+                    .optionalPaxos(ImmutablePaxosRuntimeConfiguration.builder()
+                            .leaderPingResponseWaitMs(paxos.leaderPingResponseWaitMs())
+                            .maximumWaitBeforeProposalMs(paxos.maximumWaitBeforeProposalMs())
+                            .pingRateMs(paxos.pingRateMs())
+                            .build())
+                    .slowLockLogTriggerMillis(config.slowLockLogTriggerMillis())
+                    .build();
+        } else if (TimestampBoundStoreConfiguration.class.isInstance(config.algorithm())) {
+            TimestampBoundStoreConfiguration timestampBoundStoreConfiguration =
+                    (TimestampBoundStoreConfiguration) config.algorithm();
+            install = ImmutableTimeLockInstallConfiguration.builder()
+                    .cluster(ImmutableClusterConfiguration.builder()
+                            .cluster(ServiceConfiguration.builder()
+                                    .security(timestampBoundStoreConfiguration.sslConfiguration())
+                                    .uris(config.cluster().servers())
+                                    .build())
+                            .localServer(config.cluster().localServer())
+                            .build())
+                    .asyncLock(config.asyncLockConfiguration())
+                    .optionalKvsConfig(((TimestampBoundStoreConfiguration) config.algorithm()).kvsConfig())
+                    .build();
 
-        TimeLockRuntimeConfiguration runtime = ImmutableTimeLockRuntimeConfiguration.builder()
-                .paxos(ImmutablePaxosRuntimeConfiguration.builder()
-                        .leaderPingResponseWaitMs(paxos.leaderPingResponseWaitMs())
-                        .maximumWaitBeforeProposalMs(paxos.maximumWaitBeforeProposalMs())
-                        .pingRateMs(paxos.pingRateMs())
-                        .build())
-                .slowLockLogTriggerMillis(config.slowLockLogTriggerMillis())
-                .build();
+            runtime = ImmutableTimeLockRuntimeConfiguration.builder()
+                    .slowLockLogTriggerMillis(config.slowLockLogTriggerMillis())
+                    .build();
+        } else {
+            throw new RuntimeException("The algorithm config was expected to be of type PaxosConfiguration "
```

Comment:
nit: Generally I prefer saying `unknown config type: X`, in case we add more types in the future.

Reply:
Done.